include/vcpkg/base/optional.h

```diff
@@ -74,8 +74,8 @@ namespace vcpkg
                 }
                 else if (!m_is_present && o.m_is_present)
                 {
-                    m_is_present = true;
                     new (&m_t) T(o.m_t);
+                    m_is_present = true;
```

Comment:
While we generally don't use exceptions, I think if an exception is thrown it's more appropriate to leave it as not present so the destructor won't be called.